src/rectangle.rs

```diff
@@ -32,3 +89,46 @@ impl PartialEq for Rect {
         self.ll == other.ll && self.ur == other.ur
     }
 }
+
+/// The paint mode of a [`Rect`].
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum Mode {
+    /// The shape is filled.
+    #[default]
+    Fill,
+
+    /// The shape is stroked.
+    Stroke,
+
+    /// The shape is both filled and stroked.
+    FillStroke,
+
+    /// The shape is not painted, and it is used to clip other paths.
+    Clip(
```

Comment:
You could also have a `FillStroke` with a `NonZero` and `EvenOdd` winding, so only adding `Winding` on the `Clip` is wrong, mode and winding are completely separate and can be combined in creative ways.

Reply:
You are right, but at that point I need to introduce all the possible combinations, which are not currently supported for the polygon as well.